### PR TITLE
Replace PageableManagerInterface in favor for PageableInterface

### PR DESCRIPTION
**Pull request by wbloszyk**
## Subject

I am targeting this branch, because this change should be done here. They provide little BC-break.

Closes #1871.

## Changelog

```markdown
### Changed
- Change using `Sonata\Doctrine\Model\PageableManagerInterface` in favor of `Sonata\DatagridBundle\Pager\PageableInterface`
### Removed
- Removed support for `sonata-project/datagrid-bundle` < 3.0
```

### CODE REVIEW

**Review by VincentLanglet — COMMENTED** [review]
Can you explain why the typehint are added/needed ?

**Comment by wbloszyk** [comment]
> Can you explain why the typehint are added/needed ?

Becouse `*Manager`s are implementing  `Sonata\Doctrine\Model\PageableManagerInterface` for pager which is deprecated in favor of `Sonata\DatagridBundle\Pager\PageableInterface`. In DatagridBundle 2.x all will be work. BC-break is present to allow bump DatagridBundle to v3. Otherwise update some sonata bundles will be impossible. (This which currently support only DatagridBundle v3).

**Comment by VincentLanglet** [comment]
`sonata-project/datagrid-bundle: ^3` was already allowed by the composer.json.
Do you mean it was not possible to install it ?

**Comment by wbloszyk** [comment]
> `sonata-project/datagrid-bundle: ^3` was already allowed by the composer.json.
> Do you mean it was not possible to install it ?

`Sonata\Doctrine\Model\PageableManagerInterface::getPager()` will allow for both `DatagridBundle` version (^2.0 || ^3.0)
`Sonata\DatagridBundle\Pager\PageableInterface::getPager()` is without type hints is in 2.x
`Sonata\DatagridBundle\Pager\PageableInterface::getPager()` is with type hints is in 3.x

When we change using `Sonata\Doctrine\Model\PageableManagerInterface` in favor of `Sonata\DatagridBundle\Pager\PageableInterface` we have to drop support for DatagridBundle 2.x.

File: UPGRADE-3.x.md [hunk 1]
```diff
@@ -4,6 +4,21 @@ UPGRADE 3.x
 UPGRADE FROM 3.x to 3.x
 =======================
 
+### Remove deprecated calls
+
+- `Sonata\Doctrine\Model\PageableManagerInterface` is no longer used in profit of `Sonata\DatagridBundle\Pager\PageableInterface`
+
+### Dependencies
+
+- drop support for `sonata-project/datagrid-bundle` < ^3.0.
```
Comment (phansys) [inline UPGRADE-3.x.md:13]:
```suggestion
- drop support for `sonata-project/datagrid-bundle` < 3.0.
```

File: UPGRADE-3.x.md [hunk 2]
```diff
@@ -4,6 +4,21 @@ UPGRADE 3.x
 UPGRADE FROM 3.x to 3.x
 =======================
 
+### Remove deprecated calls
+
+- `Sonata\Doctrine\Model\PageableManagerInterface` is no longer used in profit of `Sonata\DatagridBundle\Pager\PageableInterface`
```
Comment (phansys) [inline UPGRADE-3.x.md:9]:
```suggestion
- `Sonata\Doctrine\Model\PageableManagerInterface` is no longer used in profit of `Sonata\DatagridBundle\Pager\PageableInterface`.
```

File: UPGRADE-3.x.md [hunk 3]
```diff
@@ -4,6 +4,21 @@ UPGRADE 3.x
...
+
+### Dependencies
+
+- drop support for `sonata-project/datagrid-bundle` < ^3.0.
+
+    If you are extending these method you MUST add argument and return type declarations:
```
Comment (phansys) [inline UPGRADE-3.x.md:15]: BTW, we always consider this kind of change as a BC break, thus this PR should target `master`.
```suggestion
    If you are extending these methods, you MUST add argument and return type declarations:
```

**Review by phansys — CHANGES_REQUESTED** [review]

File: UPGRADE-3.x.md [hunk 4]
```diff
@@ -4,6 +4,21 @@ UPGRADE 3.x
...
+
+    - `Sonata\MediaBundle\Entity\GalleryManager::getPager()`
+    - `Sonata\MediaBundle\Entity\MediaManager::getPager()`
+    - `Sonata\MediaBundle\Model\MediaManager::getPager()`
+    - `Sonata\MediaBundle\Model\MediaManager::getPager()`
```
Comment (phansys) [inline UPGRADE-3.x.md:20]: Why is this method mentioned twice?

Comment (wbloszyk) [reply UPGRADE-3.x.md:20]: One if for entity, one is for document.

Comment (VincentLanglet) [reply UPGRADE-3.x.md:20]: Should be ``Sonata\MediaBundle\Document\MediaManager::getPager()`` then ?

There is no `Sonata\MediaBundle\Model\MediaManager`

Comment (wbloszyk) [reply UPGRADE-3.x.md:20]: Oh, correct. I check it in phone and miss mediamanager is twice

Comment (wbloszyk) [reply UPGRADE-3.x.md:20]: Done.

File: UPGRADE-3.x.md [hunk 5]
```diff
@@ -4,6 +4,21 @@ UPGRADE 3.x
 UPGRADE FROM 3.x to 3.x
 =======================
 
+### Remove deprecated calls
+
+- `Sonata\Doctrine\Model\PageableManagerInterface` is no longer used in profit of `Sonata\DatagridBundle\Pager\PageableInterface`.
```
Comment (phansys) [inline UPGRADE-3.x.md:9]:
```suggestion
### Sonata\DatagridBundle\Pager\PageableInterface

- Usages of `Sonata\Doctrine\Model\PageableManagerInterface` were replaced in favor of `Sonata\DatagridBundle\Pager\PageableInterface`.
```

File: UPGRADE-3.x.md [hunk 6]
```diff
@@ -4,6 +4,21 @@ UPGRADE 3.x
...
+
+### Dependencies
+
+- drop support for `sonata-project/datagrid-bundle` < 3.0.
```
Comment (phansys) [inline UPGRADE-3.x.md:13]:
```suggestion
- Drop support for `sonata-project/datagrid-bundle` < 3.0.
```

**Review by phansys — CHANGES_REQUESTED** [review]

**Review by phansys — COMMENTED** [review]
IMO, breaking the signature of public methods is considered a BC break.
Is there a strong reason to introduce this break in `3.x`?

**Comment by wbloszyk** [comment]
> IMO, breaking the signature of public methods is considered a BC break.
> Is there a strong reason to introduce this break in `3.x`?

Probably of overriding these mehods are very low. I want to reduce deprecation. I can also move this PR to master and do not display deprecations if people do not override it.

**Comment by VincentLanglet** [comment]
If this is a blocking point for some needed support/fixing something. We could think about it.
If this is only for a deprecation it would be better on master indeed.

**Comment by wbloszyk** [comment]
Close in favor of #1925

**Pull request closed without merging**